--- conf.d/10-auth.conf	2019-01-23 12:05:58.000000000 +0200
+++ conf.d/10-auth.conf	2019-03-04 11:54:04.477652365 +0200
@@ -48,7 +48,7 @@
 # the standard variables here, eg. %Lu would lowercase the username, %n would
 # drop away the domain if it was given, or "%n-AT-%d" would change the '@' into
 # "-AT-". This translation is done after auth_username_translation changes.
-#auth_username_format = %Lu
+auth_username_format = %Ln

 # If you want to allow master users to log in by specifying the master
 # username within the normal username string (ie. not using SASL mechanism's
@@ -119,9 +119,9 @@
 #!include auth-deny.conf.ext
 #!include auth-master.conf.ext

-!include auth-system.conf.ext
+#!include auth-system.conf.ext
 #!include auth-sql.conf.ext
-#!include auth-ldap.conf.ext
+!include auth-ldap.conf.ext
 #!include auth-passwdfile.conf.ext
 #!include auth-checkpassword.conf.ext
 #!include auth-vpopmail.conf.ext
--- conf.d/20-lmtp.conf	2019-01-23 12:05:58.000000000 +0200
+++ conf.d/20-lmtp.conf	2019-03-04 13:25:51.061844327 +0200
@@ -20,7 +20,13 @@
 # when a mail has multiple recipients.
 #lmtp_hdr_delivery_address = final
 
+recipient_delimiter =+
+lmtp_save_to_detail_mailbox = yes
+lda_mailbox_autocreate = yes
+lda_mailbox_autosubscribe = yes
+
+
 protocol lmtp {
   # Space separated list of plugins to load (default is global mail_plugins).
-  #mail_plugins = $mail_plugins
+  mail_plugins = $mail_plugins sieve
 }
--- conf.d/90-sieve.conf	2019-03-05 21:15:58.905790954 +0200
+++ conf.d/90-sieve.conf	2019-03-05 21:19:18.160059384 +0200
@@ -36,7 +38,7 @@
   # active script symlink is located.
   # For other types: use the ';name=' parameter to specify the name of the
   # default/active script.
-  sieve = file:~/sieve;active=~/.dovecot.sieve
+  sieve = /var/lib/dovecot/sieve.d/%Ln.sieve
 
   # The default Sieve script when the user has none. This is the location of a
   # global sieve script file, which gets executed ONLY if user's personal Sieve
@@ -73,7 +75,7 @@
   # to the script execution sequence in the specified order. Reading the
   # numbered sieve_before settings stops at the first missing setting, so no
   # numbers may be skipped.
-  #sieve_before = /var/lib/dovecot/sieve.d/
+  sieve_before = /var/lib/dovecot/sieve.d/before.sieve
   #sieve_before2 = ldap:/etc/sieve-ldap.conf;name=ldap-domain
   #sieve_before3 = (etc...)
 
--- conf.d/10-logging.conf	2019-03-05 21:43:01.097279575 +0200
+++ conf.d/10-logging.conf	2019-03-05 22:17:49.041027765 +0200
@@ -4,7 +4,7 @@

 # Log file to use for error messages. "syslog" logs to syslog,
 # /dev/stderr logs to stderr.
-#log_path = syslog
+log_path = /dev/stderr

 # Log file to use for informational messages. Defaults to log_path.
 #info_log_path =
@@ -41,7 +41,7 @@
 #log_core_filter =

 # Log unsuccessful authentication attempts and the reasons why they failed.
-#auth_verbose = no
+auth_verbose = yes

 # In case of password mismatches, log the attempted password. Valid values are
 # no, plain and sha1. sha1 can be useful for detecting brute force password
--- dovecot.conf	2019-04-29 23:35:05.000000000 +0300
+++ dovecot.conf	2019-05-12 00:20:17.908388214 +0300
@@ -75,7 +75,7 @@
 # Space separated list of environment variables that are preserved on Dovecot
 # startup and passed down to all of its child processes. You can also give
 # key=value pairs to always set specific settings.
-#import_environment = TZ
+import_environment = $import_environment LDAP_URIS LDAP_BASEDN TLS_KEY_FILE TLS_CERT_FILE

 ##
 ## Dictionary server settings
--- dovecot-ldap.conf.ext	2019-01-23 12:05:58.000000000 +0200
+++ dovecot-ldap.conf.ext	2019-06-29 19:09:54.704490949 +0300
@@ -21,7 +21,7 @@
 
 # LDAP URIs to use. You can use this instead of hosts list. Note that this
 # setting isn't supported by all LDAP libraries.
-#uris = 
+uris = %{env:LDAP_URIS}
 
 # Distinguished Name - the username used to login to the LDAP server.
 # Leave it commented out to bind anonymously (useful with auth_bind=yes).
@@ -66,7 +66,7 @@
 # The pass_filter is used to find the DN for the user. Note that the pass_attrs
 # is still used, only the password field is ignored in it. Before doing any
 # search, the binding is switched back to the default DN.
-#auth_bind = no
+auth_bind = yes
 
 # If authentication binding is used, you can save one LDAP request per login
 # if users' DN can be specified with a common template. The template can use
@@ -89,7 +89,7 @@
 
 # LDAP base. %variables can be used here.
 # For example: dc=mail, dc=example, dc=org
-base =
+base = %{env:LDAP_BASEDN}
 
 # Dereference: never, searching, finding, always
 #deref = never
@@ -127,8 +127,8 @@
 # userdb prefetch instead of userdb ldap in dovecot.conf. In that case you'll
 # also have to include user_attrs in pass_attrs field prefixed with "userdb_"
 # string. For example:
-#pass_attrs = uid=user,userPassword=password,\
-#  homeDirectory=userdb_home,uidNumber=userdb_uid,gidNumber=userdb_gid
+pass_attrs = uid=user,userPassword=password,\
+  homeDirectory=userdb_home,uidNumber=userdb_uid,gidNumber=userdb_gid
 
 # Filter for password lookups
 #pass_filter = (&(objectClass=posixAccount)(uid=%u))
--- conf.d/10-master.conf	2019-06-30 15:30:20.955944241 +0300
+++ conf.d/10-master.conf	2019-06-30 15:30:20.959944142 +0300
@@ -19,8 +19,8 @@
     #port = 143
   }
   inet_listener imaps {
-    #port = 993
-    #ssl = yes
+    port = 993
+    ssl = yes
   }
 
   # Number of connections to handle before starting a new process. Typically
@@ -53,15 +53,15 @@
 
 service lmtp {
   unix_listener lmtp {
-    #mode = 0666
+    mode = 0666
   }
 
   # Create inet listener only if you can't use the above UNIX socket
-  #inet_listener lmtp {
+  inet_listener lmtp {
     # Avoid making LMTP visible for the entire internet
     #address =
-    #port = 
-  #}
+    port = 25
+  }
 }
 
 service imap {
--- conf.d/10-ssl.conf	2019-07-27 09:10:44.307318844 +0300
+++ conf.d/10-ssl.conf	2020-06-07 09:47:53.412079557 +0300
@@ -9,8 +9,8 @@
 # dropping root privileges, so keep the key file unreadable by anyone but
 # root. Included doc/mkcert.sh can be used to easily generate self-signed
 # certificate, just make sure to update the domains in dovecot-openssl.cnf
-ssl_cert = </etc/dovecot/private/dovecot.pem
-ssl_key = </etc/dovecot/private/dovecot.key
+ssl_cert = </etc/ssl/certs/ssl-cert-snakeoil.pem
+ssl_key = </etc/ssl/private/ssl-cert-snakeoil.key
 
 # If key file is password protected, give the password here. Alternatively
 # give it when starting dovecot with -p parameter. Since this file is often
@@ -57,6 +57,7 @@
 #ssl_cipher_list = ALL:!kRSA:!SRP:!kDHd:!DSS:!aNULL:!eNULL:!EXPORT:!DES:!3DES:!MD5:!PSK:!RC4:!ADH:!LOW@STRENGTH
 # To disable non-EC DH, use:
 #ssl_cipher_list = ALL:!DH:!kRSA:!SRP:!kDHd:!DSS:!aNULL:!eNULL:!EXPORT:!DES:!3DES:!MD5:!PSK:!RC4:!ADH:!LOW@STRENGTH
+ssl_cipher_list=!kRSA:!3DES:!RC4:!DES:!MD5:!aNULL:!NULL:AESGCM+ECDH:AES256+ECDH:AES128:+SHA1
 
 # Colon separated list of elliptic curves to use. Empty value (the default)
 # means use the defaults from the SSL library. P-521:P-384:P-256 would be an
--- conf.d/10-mail.conf	2019-07-27 09:10:44.323318645 +0300
+++ conf.d/10-mail.conf	2019-07-27 09:10:44.299318944 +0300
@@ -27,7 +27,7 @@
 #
 # <doc/wiki/MailLocation.txt>
 #
-mail_location = mbox:~/mail:INBOX=/var/mail/%u
+mail_location = mbox:/var/mail/%Ln:INBOX=/var/mail/%Ln/Inbox
 
 # If you need to set multiple mailbox locations or want to change default
 # namespace settings, you can do it by defining namespace sections.
--- conf.d/20-imap.conf	2020-12-22 13:26:52.000000000 +0000
+++ conf.d/20-imap.conf	2022-12-05 07:33:47.876334736 +0000
@@ -95,5 +95,5 @@
 
   # Maximum number of IMAP connections allowed for a user from each IP address.
   # NOTE: The username is compared case-sensitively.
-  #mail_max_userip_connections = 10
+  mail_max_userip_connections = 15
 }
